src/basal_mass_balance/BMB_laddie.f90

```diff
@@ -0,0 +1,143 @@
+MODULE BMB_laddie
+
+  ! LADDIE model
+  ! Lambert et al. (2023) doi: 10.5194/tc-17-3203-2023
+  ! To use this option, get the code here: https://github.com/erwinlambert/laddie
+
+! ===== Preamble =====
+! ====================
+
+  USE precisions                                             , ONLY: dp
+  USE mpi_basic                                              , ONLY: par, sync
+  USE control_resources_and_error_messaging                  , ONLY: crash, init_routine, finalise_routine, colour_string
+  USE model_configuration                                    , ONLY: C
+  USE parameters
+  USE mesh_types                                             , ONLY: type_mesh
+  USE ice_model_types                                        , ONLY: type_ice_model
+  USE BMB_model_types                                        , ONLY: type_BMB_model
+  USE netcdf_input                                           , ONLY: read_field_from_file_2D
+
+  IMPLICIT NONE
+
+
+CONTAINS
+
+! ===== Main routines =====
+! =========================
+
+  SUBROUTINE run_BMB_model_laddie( mesh, BMB, time)
+    ! Calculate the basal mass balance
+    !
+    ! Call the external LADDIE model
+
+    IMPLICIT NONE
+
+    ! In/output variables:
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_BMB_model),                   INTENT(INOUT) :: BMB
+    REAL(dp),                               INTENT(IN)    :: time
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'run_BMB_model_laddie'
+    CHARACTER(LEN=256)                                    :: filename_BMB_laddie_output
+    CHARACTER(LEN=256)                                    :: laddieready
+    LOGICAL                                               :: found_laddie_file
+
+    ! Add routine to path
+    CALL init_routine( routine_name)
+
+    IF (time > C%start_time_of_run) THEN
+
+      ! Define local variables, paths to resulting melt pattern from laddie (BMB_latest.nc), and to file indicating whether laddie is ready (laddieready)
+      filename_BMB_laddie_output      = TRIM(C%fixed_output_dir) // '/laddie_output/output_BMB.nc'
```

Comment:
That's covered in the LADDIE config file. What I do is create these two config files simultaneously from templates at the start of the run, putting the runname (and UFEMISM rundirectory) into LADDIE's config file.

Indeed if this is done incorrectly, `output_BMB.nc` will remain unchanged and both models will continue running without UFEMISM reading the updated BMB file LADDIE creates. Do you want to build in a check for this? (i.e. that `BMB is not BMB_prev`?

This cannot be trivially done by trying to read the appropriate timeslice of `output_BMB.nc`, because LADDIE doesn't provide that. And it doesn't accumulate time-variable BMB into one file. So this would all take a bit of rearrangement.

Reply:
Nah, I don't think that check is needed (and might be unsafe, if you're doing a steady-state experiment and the goal is to get d[everything]/dt = 0).